clientside/src/features/board/components/Ship.tsx

```diff
@@ -0,0 +1,118 @@
+import { FC } from "react";
+import styled from "styled-components";
+import {
+    BattleshipDirection,
+    BattleshipPartType,
+    BattleshipType,
+} from "../types/battleship";
+
+export interface Props {
+    size?: number;
+    color?: string;
+    part: BattleshipPartType;
+    direction: BattleshipDirection;
+    type?: BattleshipType;
+}
+
+const ShipPart: FC<Props> = ({
+    part,
+    direction,
+    size = 32,
+    color = "#a9d5fd",
+    type = BattleshipType.Default,
+}) => {
+    let rotateDeg = 0;
+    switch (direction) {
+        case BattleshipDirection.Horizontal:
+            rotateDeg = 270;
+            break;
+        case BattleshipDirection.VerticalRev:
+            rotateDeg = 180;
+            break;
+        case BattleshipDirection.HorizontalRev:
+            rotateDeg = 90;
+            break;
+    }
+
+    switch (part) {
+        case BattleshipPartType.Front:
+            return (
+                <Wrapper rotate={rotateDeg}>
```

Comment:
Wrapper here is repeated so if you want you could extract that into one return statement